File: Prime tests/sieve.py
from math import *
from gmpy2 import is_prime
import time

def isPrime(n: int):
    A = [True for i in range(2, n + 1)]
    for i in range(2, ceil(sqrt(n))):
        if A[i - 2]:
            j = 0
            while i**2 + j*i <= n:
                A[i**2 + j*i - 2] = False
                j += 1
                if not A[n - 2]: return False
    return True

t1 = time.time()

print(isPrime(1000000))

t2 = time.time()

print(t2 - t1)

t1 = time.time()

print(is_prime(1000000))

t2 = time.time()

print(t2 - t1)

File: C++/sieve.cpp
#include <iostream>
#include <cmath>
#include <vector>

using namespace std;

bool isPrime(int n) {
    vector<bool> A(n + 1, true);
    for (int i = 2; i <= ceil(sqrt(n)); ++i) {
        if (A[i - 2]) {
            int j = 0;
            while (i * i + j * i <= n) {
                A[i * i + j * i - 2] = false;
                j++;
                if (!A[n - 2]) return false;
            }
        }
    }
    return true;
}

int main() {
    cout << isPrime(1000000);
    return 0;
}